packages/server/postgres/migrations/1649750150687_GitLabDimensionFieldMap.ts

```diff
@@ -0,0 +1,27 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+  CREATE TABLE IF NOT EXISTS "GitLabDimensionFieldMap" (
+    "id" SERIAL,
+    "teamId" VARCHAR(100) NOT NULL,
+    "dimensionName" VARCHAR(120) NOT NULL,
+    "gid" VARCHAR(140) NOT NULL,
```

Comment:
-2 I don't understand why we store the voting dimension by issue id. I think we should have the `providerId`, `projectId` instead, so it is set for the project and all upcoming voting issues.

Reply:
For the `[Integration]DimensionFieldMap`s, my understanding is we want to store the `dimensionName`, e.g. Story Points, and `labelTemplate`, e.g. `__comment`, associated with a particular project/repo.

We do this so that the user's preferences are stored so they don't need to keep changing between "As a comment" and "Do not update", for example. 

Previously, I thought it was associated with an issue, so I was storing the `gid`. But, I now see that when I update the "As a comment" label to "As a label", in the GitHub integration, it updates the label for all other issues in that repo. 

So, I agree that I should change `gid` to `projectId` so that we store the preferences for all issues within a project in GitLab.

What I'm unclear on is why we should have `providerId`? Are you thinking we should have this instead of `teamId`? The same issue could exist in multiple teams. By including the `teamId`, we only update the label preference on that team.

Reply:
I would like to tie all integration settings to their provider. As these are potentially self-hosted a client could switch between two instances where both use the same ids belonging to different projects. Or teams move orgs, there are some scenarios where this could happen. We should keep the `teamId` as well.

Reply:
Storing the providerId makes sense for that self-hosted switching edge case. So you're happy for us to store teamId, projectId, dimensionName, and providerId here?

Reply:
As the key/unique constraint to lookup the setting, yes. And then "labelTemplate" to store the actual setting.

Reply:
Sounds good. Update pushed!